docusaurus/docs/tutorials/build-a-logs-data-source-plugin.md

```diff
@@ -451,57 +450,50 @@ export class ExampleDatasource
     }
   }
 
-  // Returns an observable that will be used to fetch supplementary data based on the provided
-  // supplementary query type and original request.
-  getDataProvider(
+  // It generates a DataQueryRequest for a specific supplementary query type.
+  // @returns A DataQueryRequest for the supplementary queries or undefined if not supported.
+  getSupplementaryRequest(
     type: SupplementaryQueryType,
-    request: DataQueryRequest<ExampleQuery>
-  ): Observable<DataQueryResponse> | undefined {
+    request: DataQueryRequest<ExampleQuery>,
+    options?: SupplementaryQueryOptions
+  ): DataQueryRequest<ExampleQuery> | undefined {
     if (!this.getSupportedSupplementaryQueryTypes().includes(type)) {
       return undefined;
     }
 
     switch (type) {
       case SupplementaryQueryType.LogsVolume:
-        return this.getLogsVolumeDataProvider(request);
+        const logsVolumeOption: LogsVolumeOption =
+          options?.type === SupplementaryQueryType.LogsVolume ? options : { type };
```

Comment:
this formatting looks a bit weird, is that how it is supposed to be?

Reply:
Matches https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/datasource.ts#L183 . It becomes a really long line otherwise, but it's not a big deal.

Reply:
Looks okay, right? https://grafana.com/developers/plugin-tools/tutorials/build-a-logs-data-source-plugin#show-full-range-logs-volume